ui/components/PromptComponent.js

```diff
@@ -116,6 +132,20 @@ class PromptComponent extends React.Component {
                   {subtitle}
                 </Typography>
               </DialogContentText>
+              {showCheckbox && (
+                <FormControlLabel
+                  control={
+                    <Checkbox
+                      checked={isChecked}
+                      onChange={() => {
+                        this.handleCheckboxChange();
```

Comment:
This extra function isn't required

Reply:
I created this function to make things clean, although we can update state directly here, also Is that a good practice as clean code?

Reply:
```
onChange={this.handleCheckboxChange}
```
Isn't simple?

Reply:
Yes it is thanks for that last time actually I was doing multiple things on that so I added callback function and forgot to remove, now I get what you are talking about, it make sense